hardware/raspberrypi/bcm2711_bootloader_config.md

```diff
@@ -0,0 +1,115 @@
+# Pi4 Bootloader Configuration
+
+You can display the current configuration using 
+```
+vcgencmd bootloader_config
+```
+
+To change these bootloader configuration items, you need to extract the configuration segment, make changes, re-insert it, then reprogram the EEPROM with the new bootloader. 
+
+```
+# Extract the configuration file
+cp /lib/firmware/raspberrypi/bootloader/beta/pieeprom-2019-09-10.bin pieeprom.bin
+rpi-eeprom-config pieeprom.bin > bootconf.txt
+
+# Edit the configuration using a text editor e.g. nano bootconf.txt
+
+# If you have a UART cable then setting BOOT_UART=1 will help debug boot issues
+BOOT_UART=1
+
+# Save the new configuration and exit editor
+
+# Apply the configuration change to the EEPROM image file
+rpi-eeprom-config --out pieeprom-new.bin --config bootconf.txt pieeprom.bin
+```
+
+To update the bootloader EEPROM with the edited bootloader:
+
+```
+# Flash the bootloader EEPROM
+# Run 'rpi-eeprom-update -h' for more information
+sudo rpi-eeprom-update -d -f ./pieeprom-new.bin
+sudo reboot
+```
+
+## Configuration Items
+
+This section describes all the configuration items available in the bootloader.
+
+
+### BOOT_UART
+
+If 1 then enable UART debug output on GPIO 14 and 15. Configure the debug terminal at 115200bps, 8 bits, no parity bits, 1 stop bit. 
+Default: 0  
+Version: All  
+
+### WAKE_ON_GPIO 
+
+If 1 then 'sudo halt' will run in a lower power mode until either GPIO3 or GLOBAL_EN are shorted to ground.  
+
+Default: 0 in original version of bootloader (2019-05-10). Newer bootloaders have this set to 1.  
+Version: All  
+
+### POWER_OFF_ON_HALT  
+
+If 1 and WAKE_ON_GPIO=0 then switch off all PMIC outputs in halt. This is lowest possible power state for halt but may cause problems with some HATs because 5V will still be on. GLOBAL_EN must be shorted to ground to boot.  
+
+Default: 0  
+Version: 2019-07-15  
+
+### FREEZE_VERSION
+
+If 1 then the `rpi-eeprom-update` will skip automatic updates on this board. The parameter is not processed by the EEPROM bootloader or recovery.bin since there is no way in software of fully write protecting the EEPROM. Custom EEPROM update scripts must also check for this flag.
+
+Default: 0  
+Version: All  
+
+### BOOT_ORDER
+The BOOT_ORDER setting allows flexible configuration for the priority of different
+bootmodes. It is represented as 32bit unsigned integer where each nibble represents
+a bootmode. The bootmodes are attempted in LSB to MSB order.  
```

Comment:
It might be worth explaining "LSB to MSB order" (or using some other phrasing) for the benefit of non-techies?

Reply:
Rephrased.